src/ias_client/src/ias_client.rs

```diff
@@ -0,0 +1,230 @@
+    }
+
+    pub fn set_timeout(&mut self, timeout: Duration) {
+        self.timeout = timeout;
+    }
+
+    /// Get request to receive signature revocation list for input Group ID (gid). Accepts
+    /// optional 'gid' and optional 'api_path' as input. Optional 'gid' field is used for the
+    /// case of IAS Proxy server, which receives request with 'gid' appended already.
+    ///
+    /// return: A ClientResponse object containing the following:
+    ///     Body of the response has 'signature revocation list', the body of the response from ISA.
+    ///     Header of the response has nothing.
+    pub fn get_signature_revocation_list(
+        &self,
+        gid: Option<&str>,
+        api_path: Option<&str>,
+    ) -> Result<ClientResponse, ClientError> {
+
+        // Path to get SigRL from
+        let mut final_path = String::new();
+        final_path.push_str(self.ias_url.as_str());
+        // Received REST path if any
+        let received_path = match api_path {
+            Some(path_present) => path_present,
+            _ => SIGRL_LINK,
+        };
+        final_path.push_str(received_path);
+        // Append gid to the path if present
+        let received_gid = match gid {
+            Some(gid_present) => gid_present,
+            _ => "",
+        };
+        final_path.push_str(received_gid);
+        let url = final_path.parse::<Uri>().expect("Error constructing URI from string");
+        debug!("Fetching SigRL from: {}", url);
+
+        // Send request to get SigRL
+        let client = get_client(self.spid_cert.as_slice())
+            .expect("Eror creating http/s client");
+        // TODO: Add logic for request timeout
+        let response = client.get(url);
+        read_response_future(response)
+    }
+
+    /// Post request to send Attestation Enclave Payload and get response having Attestation
+    /// Verification Report. Accepts quote and optional values pse_manifest, nonce as input.
+    ///
+    /// return: A ClientResponse object containing the following:
+    ///     Body of the response has 'attestation verification report', the body (JSON) of the
+    ///         response from ISA.
+    ///     Header of the response has 'signature', the base 64-encoded RSA-SHA256 signature of the
+    ///         response body (aka, AVR) using the report key.
+    pub fn post_verify_attestation(
+        &self,
+        quote: &[u8],
+        manifest: Option<&str>,
+        nonce: Option<u64>,
+    ) -> Result<ClientResponse, ClientError> {
+
+        // REST API to connect to for getting AVR
+        let mut final_path = String::new();
+        final_path.push_str(self.ias_url.as_str());
+        final_path.push_str(AVR_LINK);
+        let url = final_path.parse::<Uri>().expect("Error constructing URI from string");
+        debug!("Posting attestation verification request to: {}", url);
+
+        // Construct AEP, request parameter
+        let mut request_aep: HashMap<String, String> = HashMap::new();
```

Comment:
Can we use struct instead of HashMap

Reply:
I will add a comment describing why not using a struct at this time.